### ISSUE TITLE
Add support for Pulsar Editor

### ISSUE THREAD

**Issue by mesub7** [HelloWorld017/atom-discord#150]
As I'm sure you know, Atom was sunsetted a while back and a community-maintained replacement (Pulsar) has been available for a while. I don't suppose you could update the package to match this, please?

**Comment by ShadowLp174** [comment]
Not the maintainer but I don't really expect any updates on here 🙃

Though, a few days ago, @siliconsniffer created a fork that's supposed to do what you're asking for (I'm not sure if it's completed already)

Link: https://github.com/siliconsniffer/pulsar-discord

Hope this helps, if you have any further questions just let me know

**Comment by siliconsniffer** [comment]
> Not the maintainer but I don't really expect any updates on here 🙃
> 
> Though, a few days ago, @siliconsniffer created a fork that's supposed to do what you're asking for (I'm not sure if it's completed already)
> 
> Link: https://github.com/siliconsniffer/pulsar-discord
> 
> Hope this helps, if you have any further questions just let me know

Nope still WIP, not much progress done yet. If you want to, I can add you or @mesub7 to the repo.
As far as I can see we need to change the details (e.g. Atom to Pulasr) in the json-files under /i18n.
But I have no idea how the atom extensions work...

**Comment by ShadowLp174** [comment]
> > Not the maintainer but I don't really expect any updates on here 🙃
> > 
> > Though, a few days ago, @siliconsniffer created a fork that's supposed to do what you're asking for (I'm not sure if it's completed already)
> > 
> > Link: https://github.com/siliconsniffer/pulsar-discord
> > 
> > Hope this helps, if you have any further questions just let me know
> 
> Nope still WIP, not much progress done yet. If you want to, I can add you or @mesub7 to the repo.
> As far as I can see we need to change the details (e.g. Atom to Pulasr) in the json-files under /i18n.
> But I have no idea how the atom extensions work...

Hmm I see
I'm not sure if modifying the localisations will be enough, there may be hard coded values - we'll see ig

**Comment by sparklost** [comment]
Yes, you are correct, there is one almost hardcoded thing that needs to be changed:
It is [here](https://github.com/HelloWorld017/atom-discord/blob/master/src/config.js#L208).
This is default value for Discord app ID. This app contains that "Playing Atom Editor" string and all images.
Someone should create new app for Pulsar with updated images and "Playing Pulsar Edit".
Creating custom app is explained [in FAQ](https://github.com/HelloWorld017/atom-discord/blob/master/FAQ.md#playing-atom-editor-text).

Also, to be sure every atom-discord string is replaced, here are commands for linux, run them in project directory:
```sh
find ./ -type f -exec sed -i 's/atom-discord/pulsar-discord/g' {} \;
find ./ -type f -exec sed -i 's/Atom-Discord/Pulsar-Discord/g' {} \;
```

It would be nice to see pulsar-discord on https://web.pulsar-edit.dev/

**Comment by siliconsniffer** [comment]
I've abandoned this as I've lost interest in Pulsar. The team behind Atom started another editor called [Zed](https://zed.dev) which some of you might be interested in. There is also already a discord plugin for it.

**Comment by mesub7** [comment]
Zed doesn't support windows yet, which is disappointing.

Is your repo still up? Somebody may want to fork it.

mesub

**Comment by siliconsniffer** [comment]
Yep, Windows support is still WIP. I've deleted my repo because I haven't worked on it that much, so there is hardly any progress lost.

**Comment by sparklost** [comment]
Zed has not nearly enough extensions as Pulsar (or even vscode), eg. minimap, cursor history, ask stack... and I need them.
Its nice that it is so fast, but still wont fill all my needs as Pulsar does. Then vscode also miss some great extensions, customizability and its not privacy oriented.
But we are not here to discuss which editor is better.

Anyway, guess what I just did:
![Screenshot From 2024-11-11 01-42-02](https://github.com/user-attachments/assets/71c6b2ed-3e15-4c1f-b8e6-ed2a089a82eb)

I will make fork of this repo called pulsar-discord, with modified readme, FAQ, settings and code. And publish package on https://web.pulsar-edit.dev/

**Comment by sparklost** [comment]
YAY its live! 🎉 
https://github.com/mzivic7/pulsar-discord
https://web.pulsar-edit.dev/packages/pulsar-discord
Can be installed through builtin package manager.

**Comment by siliconsniffer** [comment]
Yeah, I agree that Zed is lacking quite many features other editors already have. But keep in mind how "new" this editor is. Anyways,  nice work with the plugin! Can this be closed then? Great to see you completed it.

**Comment by sparklost** [comment]
Thanks. 
Yes, I think issue can be closed now.
